src/work-allocation/guards/work-allocation-feature-toggle.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
+import { AppConstants } from 'src/app/app.constants';
+
+@Injectable()
+export class WorkAllocationFeatureToggleGuard implements CanActivate {
+    constructor(private readonly featureToggleService: FeatureToggleService
+    ) {}
+
+    public canActivate() {
```

Comment:
If you had the `canActivate()` method take the `route: ActivatedRouteSnapshot` parameter, you could set up the route with the constant instead and then get it with something like `route.data.featureName`.

Then we'd have a generic route guard for feature toggles.

Reply:
It's already agreed and discussed with @ronaldmansveld  that this will not be generic. We do have a plan to implement a generic feature toggle guard and put it in Common.Library. Not for this PR.